### ISSUE TITLE
Dev: 'Incorrect magic code. Please try again or request a new code.' is displayed in the empty email field when switching back to online after refreshing the page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#18310]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open chrome web
2. Logout if already signed in
3. Enter your email and click on continue
4. Turn off your wifi
5. Now enter any invalid code and now refresh the page to make sure that the magic code is now empty
6. Click on go back and go to the email page
7. Turn on your wifi and notice that in a few moments , it displays 'Incorrect magic code. Please try again or request a new code' in the empty email field

## Expected Result:
 'Incorrect magic code. Please try again or request a new code.' should not be displayed in the empty email field when switching back to online after refreshing the page
## Actual Result:'
 Incorrect magic code. Please try again or request a new code.' is displayed in the empty email field when switching back to online after refreshing the page
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** Dev
**Reproducible in staging?:** n/a
**Reproducible in production?:** n/a
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/235796469-00502825-5e13-4003-bbd3-708a94ea08b9.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @priya-zha
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1683030330093459

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jliexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Triggered auto assignment to @marcaaron (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by jliexpensify** [comment]
Re-assigning for Engineering eyes (as per this [comment](https://expensify.slack.com/archives/C01SKUP7QR0/p1681308282979459?thread_ts=1681291001.229719&cid=C01SKUP7QR0))

**Comment by hungvu193** [comment]
This is same root cause with https://github.com/Expensify/App/issues/18311.
The problem is still with submitting the request in offline mode and it can be solve with this proposal as well.
https://github.com/Expensify/App/issues/18311#issuecomment-1532380252

**Comment by priya-zha** [comment]
Since it has already been deployed to staging, please skip the 'Refresh the page' in step 5 of the steps mentioned above to reproduce the bug

**Comment by marcaaron** [comment]
>Enter your email and click on continue
>Turn off your wifi
>Now enter any invalid code and now refresh the page to make sure that the magic code is now empty
>Click on go back and go to the email page
>Turn on your wifi and notice that in a few moments , it displays 'Incorrect magic code. Please try again or request a new code' in the empty email field

This honestly sounds like a pretty reasonable expected behavior for this edge case. We entered an invalid code. The user went offline. They then reloaded the app for some reason. The request was retried. We showed them some error message about. It's maybe not in the most optimal place, but certainly doesn't seem like something we need to prioritize. I'm going to call this a "Feature Request" and invite someone to proposed a design change in Slack. But since this apparently only happens on "dev" and the behavior is not 100% clear - I'm closing.
